sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/subquery.scala

```diff
@@ -264,11 +264,9 @@ object PullupCorrelatedPredicates extends Rule[LogicalPlan] with PredicateHelper
       } else {
         (transformed, baseConditions)
       }
-      (plan, stripOuterReferences(deDuplicatedConditions))
```

Comment:
This diff seems dicey... This might lead to unresolved outer reference after this batch.